platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -268,10 +268,9 @@ public MerkleDbDataSource(
         final long hashesRamToDiskThreshold = tableConfig.getHashesRamToDiskThreshold();
         if (hashesRamToDiskThreshold > 0) {
             if (Files.exists(dbPaths.hashStoreRamFile)) {
-                hashStoreRam = new HashListByteBuffer(dbPaths.hashStoreRamFile, hashesRamToDiskThreshold);
+                hashStoreRam = new HashListByteBuffer(dbPaths.hashStoreRamFile, hashesRamToDiskThreshold, config);
             } else {
-                final int hashesPerBuffer = merkleDbConfig.hashStoreRamBufferSize();
-                hashStoreRam = new HashListByteBuffer(hashesPerBuffer, hashesRamToDiskThreshold, true);
+                hashStoreRam = new HashListByteBuffer(hashesRamToDiskThreshold, config);
```

Comment:
I think only config can be passed, as is a part of config -- see this: https://github.com/hashgraph/hedera-services/blob/47e2604801d5336ec8e8ee3b2c276b8f89d72097/hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/MerkleSchemaRegistry.java#L333-L333

Reply:
This is a different config, `MerkleDbConfig` vs `MerkleDbTableConfig`. You're right, the threshold is set in `MerkleSchemaRegistry`, it's passed as a `MerkleDbTableConfig` to this code, and `hashesRamToDiskThreshold` is initialized from that table config and passed to `HashListByteBuffer`